File: CS/XPO/ASP.NETCore/Blazor/RuntimeDbChooser.Blazor.Server/Services/ConnectionStringProvider.cs
using DevExpress.ExpressApp.Security;
using RuntimeDbChooser.Services;

namespace RuntimeDbChooser.Blazor.Server.Services;
public class ConnectionStringProvider : IConnectionStringProvider {
    readonly ILogonParameterProvider logonParameterProvider;
    readonly IConnectionStringHelper connectionStringHelper;

    public ConnectionStringProvider(ILogonParameterProvider logonParameterProvider, IConnectionStringHelper connectionStringHelper) {
        this.logonParameterProvider = logonParameterProvider;
        this.connectionStringHelper = connectionStringHelper;
    }

    public string GetConnectionString() {
        //Configure the connection string based on logon parameter values.
        string? targetDataBaseName = logonParameterProvider.GetLogonParameters<IDatabaseNameParameter>().DatabaseName?.Name;
        if(targetDataBaseName != null) {
            return connectionStringHelper.GetConnectionStringsMap()[targetDataBaseName];
        }
        return "not set";
    }
}

File: VB/EFCore/ASP.NETCore/Blazor/RuntimeDbChooser.Blazor.Server/Services/ConnectionStringProvider.vb
Imports DevExpress.ExpressApp.Security
Imports RuntimeDbChooser.Services
Imports System.Linq

Namespace RuntimeDbChooser.Blazor.Server.Services

    Public Class ConnectionStringProvider
        Implements RuntimeDbChooser.Services.IConnectionStringProvider

        Private ReadOnly logonParameterProvider As ILogonParameterProvider

        Private ReadOnly connectionStringHelper As RuntimeDbChooser.Services.IConnectionStringHelper

        Public Sub New(ByVal logonParameterProvider As ILogonParameterProvider, ByVal connectionStringHelper As RuntimeDbChooser.Services.IConnectionStringHelper)
            Me.logonParameterProvider = logonParameterProvider
            Me.connectionStringHelper = connectionStringHelper
        End Sub

        Public Function GetConnectionString() As String Implements RuntimeDbChooser.Services.IConnectionStringProvider.GetConnectionString
            'Configure the connection string based on logon parameter values.
            Dim targetDataBaseName As String? = logonParameterProvider.GetLogonParameters(Of RuntimeDbChooser.Services.IDatabaseNameParameter)().DatabaseName?.Name
            If targetDataBaseName IsNot Nothing Then
                Return connectionStringHelper.GetConnectionStringsMap()(targetDataBaseName)
            End If

            Return connectionStringHelper.GetConnectionStringsMap().Values.First()
        End Function
    End Class
End Namespace
